marketpulse/main/models.py

```diff
@@ -41,32 +42,29 @@ def __unicode__(self):
         return '{0}, {1}'.format(self.shop_name, self.country)
 
 
-class Plan(models.Model):
-    """Mobile phone plan information."""
-
-    has_plan = models.BooleanField(default=False)
-    duration = models.IntegerField(default=None)
-    description = models.TextField(default='', blank=True)
-
-
 class Contribution(models.Model):
     """Model for contribution data."""
 
     user = models.ForeignKey(settings.AUTH_USER_MODEL, related_name='contributions')
     activity = models.ForeignKey(Activity, related_name='contributions')
-    location = models.ForeignKey(Location, related_name='contributions')
-    device = models.ForeignKey(Device, related_name='contributions')
+    location = models.ForeignKey(Location, related_name='contributions', null=True)
+    device = models.ForeignKey(Device, related_name='contributions', null=True)
     created_on = models.DateTimeField(auto_now_add=True)
     updated_on = models.DateTimeField(auto_now=True)
     comment = models.TextField(blank=True, default='')
     availability = models.BooleanField(default=True)
-    plan = models.ManyToManyField(Plan, through='Price', null=True, default=None)
+
+    def __unicode__(self):
+        return '{0}, {1}'.format(self.user, self.activity)
 
 
-class Price(models.Model):
-    """Mobile phone price model."""
+class Plan(models.Model):
+    """Mobile phone plan information."""
 
-    contribution = models.ForeignKey(Contribution)
-    plan = models.ForeignKey(Plan)
-    amount = models.IntegerField()
-    currency = models.CharField(max_length=3, choices=[])
+    contribution = models.ForeignKey(Contribution, related_name='plans', null=True)
```

Comment:
Same here, why do we need null=True? If there is no contribution there shouldn't be any ``Plan``.